### PR TITLE
Contract stress test

### PR DESCRIPTION
**Pull request by kettlebell**

### CODE REVIEW

File: spectrum-offchain-lm/src/scheduler/process.rs [hunk 1]
```diff
@@ -43,14 +44,17 @@ where
         let network = network.clone();
         async move {
             if tip_reached.is_completed() {
-                let mut schedules = schedules.lock().await;
```
Comment (kettlebell) [inline spectrum-offchain-lm/src/scheduler/process.rs:46]: Before this PR, this lock was held for far too long, leading to deadlocks.

File: spectrum-offchain/src/event_sink.rs [hunk 2]
```diff
@@ -27,15 +26,15 @@ where
         let def_han = def_handler_arc.clone();
         async move {
             let mut unhandled_ev = Some(ev.clone());
-            let mut hans_guard = hans.lock().unwrap();
-            for han in hans_guard.iter_mut() {
+            let mut hans_guard = hans.lock().await;
```
Comment (kettlebell) [inline spectrum-offchain/src/event_sink.rs:29]: Switch to async-Mutex.

File: spectrum-offchain/src/executor.rs [hunk 3]
```diff
@@ -218,7 +218,7 @@ pub fn parse_err(err: &str) -> NodeSubmitTxError {
         return NodeSubmitTxError::MissingInputs(
             s[..s.len() - 1] // Don't consider the trailing ']' character
                 .split(',')
-                .map(|s| s.parse::<i32>().unwrap())
+                .map(|s| s.trim().parse::<i32>().unwrap())
```
Comment (kettlebell) [inline spectrum-offchain/src/executor.rs:221]: Fixed a bug where indices with surrounding whitespace wasn't getting parsed

File: spectrum-offchain-lm/src/data/order.rs [hunk 4]
```diff
@@ -397,6 +397,7 @@ impl TryFromBox for Deposit {
 #[serde(into = "RawRedeem")]
 pub struct Redeem {
     pub order_id: OrderId,
+    pub pool_id: PoolId,
```
Comment (kettlebell) [inline spectrum-offchain-lm/src/data/order.rs:400]: I didn't rebase this PR to the right spot, so you're seeing this again.

**Review by kettlebell — COMMENTED** [review]
Main issues fixed:
 - Fix deadlocks by holding Mutexes for shorter durations
 - ~More resilient order handling~

**Review by oskin1 — APPROVED** [review]

**Pull request merged**